File: java/simple-cassandra-client/src/main/java/com/example/cassandra/AsynchronousExample.java
package com.example.cassandra;

import com.datastax.driver.core.Query;
import com.datastax.driver.core.ResultSetFuture;
import com.datastax.driver.core.Row;
import com.datastax.driver.core.querybuilder.QueryBuilder;

public class AsynchronousExample extends SimpleClient {
	public AsynchronousExample() {
	}
	
	public ResultSetFuture getRows() {
		Query query = QueryBuilder.select().all().from("simplex", "songs");
		return getSession().executeAsync(query);
	}

	public static void main(String[] args) {
		AsynchronousExample client = new AsynchronousExample();
		client.connect("127.0.0.1");
		client.createSchema();
		client.loadData();
		ResultSetFuture results = client.getRows();
      for (Row row : results.getUninterruptibly()) {
      	System.out.printf( "%s: %s / %s\n",
      			row.getString("artist"), 
      			row.getString("title"), 
      			row.getString("album") );
      }
		client.dropSchema("simplex");
		client.close();
	}
}

File: csharp/SimpleCassandraClient/SimpleCassandraClient/AsynchronousExample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Cassandra;

namespace CassandraExamples
{
    public class AsynchronousExamples : SimpleClient
    {
        public AsynchronousExamples()
        {
        }

        public CqlRowSet getRows()
        {
            //Query query = QueryBuilder.select().all().from("simplex", "songs");
            //return session.executeAsync(query);
            Query query = new SimpleStatement("SELECT * FROM simplex.songs;");

            return session.Execute(query);
        }
    }
}
